cmd/containerd-shim-runhcs-v1/service_internal.go

```diff
@@ -146,6 +156,18 @@ func (s *service) createInternal(ctx context.Context, req *task.CreateTaskReques
 		spec.Windows.LayerFolders = append(spec.Windows.LayerFolders, source)
 	}
 
+	// write the layers to a file so that it can be used for proper cleanup We can't write to the
+	// config.json as it is read-only for shim.
+	f, err = os.Create(filepath.Join(req.Bundle, layersFile))
```

Comment:
Also creating the layersFile for spec.Linux and legacyMountType?

Reply:
Yeah, it will be ignored for LCOW & legacy WCOW layers. Do you think there are any issues with creating this file for other layer types as well?

Reply:
It adds additional failure modes for no benefit. It also creates confusion since the comment says that this file is used for proper cleanup but that is only true for CIMFS layers. Is there harm in scoping this to CIMFS layers?

Reply:
fixed.